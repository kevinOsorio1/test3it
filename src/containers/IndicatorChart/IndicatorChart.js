import React from 'react';
import {Text, useWindowDimensions, View, StyleSheet} from 'react-native';
import {LineChart} from 'react-native-chart-kit';
import useChart, {chartConfig} from '../../utils/hooks/useChart';

const IndicatorChart = ({indicator}) => {
  const {width} = useWindowDimensions();
  const {data, labels} = useChart(indicator);

  return (
    <LineChart
      formatYLabel={yValue => `$ ${yValue}`}
      xLabelsOffset={36}
      renderDotContent={({x, y, index, indexData}) => (
        <View key={index} style={chartStyles.dotView(y, x)}>
          <Text testID={`dot-${index}`}>{indexData}</Text>
        </View>
      )}
      width={width}
      height={500}
      data={{
        labels: labels,
        datasets: [
          {
            data: data,
            color: () => 'rgba(134, 65, 244, 1)',
            strokeWidth: 2,
          },
        ],
      }}
      chartConfig={chartConfig}
    />
  );
};

export const chartStyles = StyleSheet.create({
  dotView: (y, x) => ({
    position: 'absolute',
    top: y + 8,
    left: x - 30,
    backgroundColor: 'rgba(134, 65, 244, 0.7)',
    padding: 4,
  }),
});

export default IndicatorChart;
