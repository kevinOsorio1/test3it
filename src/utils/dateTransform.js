export const stringToDateFormat = date => {
  return new Date(date).toLocaleDateString('zh-Hans-CN');
};
