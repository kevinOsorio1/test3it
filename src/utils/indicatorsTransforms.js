export const apiToBO = apiobject => {
  return {
    name: apiobject.nombre,
    unit: apiobject.unidad_medida,
    code: apiobject.codigo,
  };
};
export const indicatorTransform = indicatorsApi => {
  const avoidKeys = ['version', 'autor', 'fecha'];
  return Object.keys(indicatorsApi).reduce((acc, el) => {
    if (avoidKeys.includes(el)) {
      return acc;
    }
    return [...acc, apiToBO(indicatorsApi[el])];
  }, []);
};

const seriesTransform = series => {
  return series.reduce((acc, el) => {
    return [...acc, {date: el.fecha, value: el.valor}];
  }, []);
};

export const oneIndicatorToBO = indicator => {
  console.log('api indicator', indicator);
  return {
    name: indicator.nombre,
    code: indicator.codigo,
    unit: indicator.unidad_medida,
    prefix: units[indicator.unidad_medida],
    date: indicator.serie[0].fecha,
    serie: [...seriesTransform(indicator.serie)],
  };
};

export const units = {
  Pesos: '$',
  Porcentaje: '%',
};
