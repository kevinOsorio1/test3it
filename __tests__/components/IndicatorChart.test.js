import React from 'react';
import {render} from '@testing-library/react-native';
import {indicator} from '../../__mocks__/indicators.mocks';
import {oneIndicatorToBO} from '../../src/utils/indicatorsTransforms';
import IndicatorChart from '../../src/containers/IndicatorChart/IndicatorChart';

test('should show indicator chart', () => {
  const rendered = render(<IndicatorChart />);
  expect(rendered.toJSON).toMatchSnapshot();
});
test('should give correct style', () => {
  const data = oneIndicatorToBO(indicator);
  const rendered = render(<IndicatorChart indicator={data} />);
  expect(rendered.toJSON).toMatchSnapshot();
});
