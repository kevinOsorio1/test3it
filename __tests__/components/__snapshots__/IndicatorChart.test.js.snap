// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should give correct style 1`] = `[Function]`;

exports[`should show indicator chart 1`] = `[Function]`;
